src/renderer/services/binance/utils.ts

```diff
@@ -81,3 +86,40 @@ export const bncSymbolToAsset = (symbol: string): O.Option<Asset> =>
  * Converts a BinanceChain symbol to an `Asset` string
  **/
 export const bncSymbolToAssetString = (symbol: string) => `BNB.${symbol}`
+
+export const toAssetWithBalances = (balancesRD: BalancesRD, intl?: IntlShape): AssetsWithBalanceRD =>
+  FP.pipe(
+    balancesRD,
+    RD.map((balances) =>
+      FP.pipe(
+        balances.map((balance) =>
+          FP.pipe(
+            bncSymbolToAsset(balance.symbol),
+            O.map<Asset, AssetWithBalance>((asset) => ({
+              asset,
+              balance: assetAmount(balance.free),
+              frozenBalance: assetAmount(balance.frozen)
+            }))
+          )
+        ),
+        sequenceTOptionFromArray
+      )
+    ),
+    RD.chain((balances) =>
+      RD.fromOption(balances, () =>
+        Error(intl?.formatMessage({ id: 'wallet.send.errors.balancesFailed' }) ?? 'Transform failed')
+      )
+    )
+  )
+
+export const getAssetWithBalance = (
+  balancesRD: AssetsWithBalanceRD,
+  asset: O.Option<Asset>
+): O.Option<AssetWithBalance> =>
+  FP.pipe(
+    asset,
+    O.chain((asset) =>
+      FP.pipe(balancesRD, RD.map(A.findFirst((assetWB) => assetWB.asset.symbol === asset.symbol)), RD.toOption)
+    ),
+    O.flatten
+  )
```

Comment:
what do think about such implementation ? 
```ts
export const getAssetWithBalance = (
  balancesRD: AssetsWithBalanceRD,
  asset: O.Option<Asset>
): O.Option<AssetWithBalance> =>
  FP.pipe(
    sequenceTOption(asset, RD.toOption(balancesRD)),
    O.chain(([asset, balances]) =>
      FP.pipe(
        balances,
        A.findFirst((balance) => balance.asset.symbol === asset.symbol)
      )
    )
  )
```
in this case we are always in terms of Option only